components/transfer/transfer-list-item.component.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  NgZone,
+  Output,
+  TemplateRef,
+  ViewEncapsulation
+} from '@angular/core';
+
+import type { TransferItem } from './interface';
+
+@Component({
+  selector: '[nz-transfer-list-item]',
+  exportAs: 'nzTransferListItem',
+  preserveWhitespaces: false,
+  template: `
+    <ng-template #renderContainer>
+      <div class="ant-transfer-list-content-item-text">
+        <ng-template [ngTemplateOutlet]="render" [ngTemplateOutletContext]="{ $implicit: item }"> </ng-template>
+      </div>
+    </ng-template>
+    <ng-template #renderDefault>
+      <label
+        nz-checkbox
+        [nzChecked]="checked"
+        (click)="clickEvt($event)"
+        (nzCheckedChange)="handleItemSelect()"
+        [nzDisabled]="disabled"
+      >
+        <ng-container *ngIf="!render; else renderContainer">
+          <div class="ant-transfer-list-content-item-text">{{ item.title }}</div>
+        </ng-container>
+      </label>
+    </ng-template>
+    <ng-container *ngIf="showRemove; else renderDefault">
+      <div class="ant-transfer-list-content-item-text">{{ item.title }}</div>
+      <div
+        class="ant-transfer-list-content-item-remove"
+        [style.pointer-events]="disabled ? 'none' : null"
+        (click)="remove.emit()"
+      >
+        <i nz-icon nzType="delete"></i>
+      </div>
+    </ng-container>
+  `,
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    class: 'ant-transfer-list-content-item',
+    '[class.ant-transfer-list-content-item-disabled]': 'disabled',
+    '[class.ant-transfer-list-content-item-checked]': 'checked',
+    '(click)': 'handleItemSelect()'
+  }
+})
+export class NzTransferListItemComponent {
+  @Input() item!: TransferItem;
+  @Input() checked?: boolean = false;
+  @Input() disabled?: boolean = false;
+  @Input() render: TemplateRef<void> | null = null;
+  @Input() showRemove: boolean = false;
+  @Output() readonly itemSelect = new EventEmitter<void>();
+  @Output() readonly remove = new EventEmitter<void>();
+
+  constructor(private ngZone: NgZone) {}
+
+  clickEvt(ev: Event): void {
+    this.ngZone.runOutsideAngular(() => ev.stopPropagation());
```

Comment:
Hey, I'll leave some comments regarding this.

The `click` event listener will still trigger the change detection, and the `stopPropagation` will be called outside of the Angular zone, a synchronous API. I find it a bit complicated for dynamic elements (that are within `ng-template`s) and how to handle their event listeners. I usually just create a behaviour subject and store the dynamic element there when it's set. Lemme show a quick example:
```ts
@Component({
  template: `
    <ng-template #renderDefault>
      <label
        nz-checkbox
        [nzChecked]="checked"
        (click)="clickEvt($event)"
        (nzCheckedChange)="handleItemSelect()"
        [nzDisabled]="disabled"
      >
        <div class="ant-transfer-list-content-item-text" *ngIf="!render; else renderContainer">
          {{ item.title }}
        </div>
      </label>
    </ng-template>
  `
})
export class NzTransferListItemComponent implements OnInit {
  @ViewChild(NzCheckboxComponent, { read: ElementRef, static: false })
  set label(label: ElementRef<HTMLElement> | undefined) {
    // The label will equal `undefined` on the first change detection cycle if there's no element
    // in the DOM.
    if (label) {
      this.label$.next(label);
    }
  }

  constructor(private ngZone: NgZone, private destroy$: NzDestroyService) {}

  private label$ = new BehaviorSubject<null | ElementRef<HTMLElement>>(null);

  ngOnInit(): void {
    this.label$
      .pipe(
        switchMap(label => {
          if (label) {
            return new Observable<MouseEvent>(subscriber =>
              this.ngZone.runOutsideAngular(() =>
                fromEvent<MouseEvent>(label.nativeElement, 'click').subscribe(subscriber)
              )
            );
          } else {
            // Do not emit anything and just remove the event listener if label has been removed
            // from the DOM.
            return EMPTY;
          }
        }),
        takeUntil(this.destroy$)
      )
      .subscribe(event => event.stopPropagation());
  }
}
```

I've already used the same approach in different places btw, and it looks like an overhead. I guess we could just create some service within `ng-zorro-antd/core/services`, e.g. `NzEventManager` (or some other name that fits its purposes):
```ts
@Injectable({ providedIn: 'root' })
export class NzEventManager {
  constructor(private ngZone: NgZone) {}

  addEventListenerOutsideAngular<T extends Event>(target: HTMLElement, eventName: string): Observable<T> {
    return new Observable<T>(subscriber =>
      this.ngZone.runOutsideAngular(() => fromEvent<T>(target as any, eventName).subscribe(subscriber))
    );
  }
}
```